### PR TITLE
:seedling: Add test scripts.

### PR DESCRIPTION
**Pull request by xuezhaojun**
## Summary

There are 2 scripts added:

Sometimes, we want to run integration/e2e test multiple times to reproduce the flaky error, this script allows you to run test many times util it's failed, then it collects then output  into` _output/flaky-error-test/` directory:

```sh
OCM git:(main) ✗ sh hack/integration_flaky_error_test.sh 10
Running test 1 of 2...
Test 1 passed
Running test 2 of 2...
...
Test 7 failed
Test 7 result stored in _output/flaky-error-test/integration_test_20240702-135926_7.output
```

---

Actually it tests out a integration and a e2e flaky error already:

```
[38;5;243m/Users/xuezhao/workspaces/OCM/test/integration/addon/addon_manager_upgrade_test.go:180[0m

  [38;5;9m[FAILED] Failed after 3.010s.
  Expected
      <*errors.errorString | 0xc000fbfca0>: 
      unexpected condition matches before duration
      {
          s: "unexpected condition matches before duration",
      }
  to be nil
```

---

```
• [FAILED] [360.103 seconds]
Enable addon management feature gate [BeforeEach] Template type addon should be functioning [addon-manager]
  [BeforeEach] /home/runner/work/ocm/ocm/test/e2e/addonmanagement_test.go:68
  [It] /home/runner/work/ocm/ocm/test/e2e/addonmanagement_test.go:171

  [FAILED] Timed out after 300.001s.
  Expected success, but got an error:
      <*errors.errorString | 0xc000a74c20>: 
      the addon e2e-managedcluster-n655ww/hello-template available condition is not true, [{Progressing True 0 2024-07-04 01:45:49 +0000 UTC Progressing progressing... work is not ready} {RegistrationApplied True 0 2024-07-04 01:45:49 +0000 UTC SetPermissionApplied Registration of the addon agent is configured} {ClusterCertificateRotated False 0 2024-07-04 01:45:49 +0000 UTC ClientCertificateUpdateFailed Failed to rotated client certificate namespaces "open-cluster-management-agent-addon" not found} {ManifestApplied False 0 2024-07-04 01:45:49 +0000 UTC AddonManifestAppliedFailed failed to apply the manifests of addon} {Available False 0 2024-07-04 01:45:49 +0000 UTC WorkNotApplied 3 of 4 resources are not available}]
      {
          s: "the addon e2e-managedcluster-n655ww/hello-template available condition is not true, [{Progressing True 0 [202](https://github.com/open-cluster-management-io/ocm/actions/runs/9787073322/job/27022975921?pr=545#step:10:203)4-07-04 01:45:49 +0000 UTC Progressing progressing... work is not ready} {RegistrationApplied True 0 2024-07-04 01:45:49 +0000 UTC SetPermissionApplied Registration of the addon agent is configured} {ClusterCertificateRotated False 0 2024-07-04 01:45:49 +0000 UTC ClientCertificateUpdateFailed Failed to rotated client certificate namespaces \"open-cluster-management-agent-addon\" not found} {ManifestApplied False 0 2024-07-04 01:45:49 +0000 UTC AddonManifestAppliedFailed failed to apply the manifests of addon} {Available False 0 2024-07-04 01:45:49 +0000 UTC WorkNotApplied 3 of 4 resources are not available}]",
      }
  In [BeforeEach] at: /home/runner/work/ocm/ocm/vendor/github.com/onsi/gomega/internal/async_assertion.go:145 @
```

I will work on this soon.

### CODE REVIEW

**Comment by qiujian16** [comment]
But I think the flaky is mainly on e2e?

**Comment by xuezhaojun** [comment]
Agree, hold the PR to add e2e script as well.
/hold

**Comment by xuezhaojun** [comment]
@qiujian16 The scripts updated, please take another look

File: hack/e2e_flaky_error_test.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+mkdir -p "_output/flaky-error-test"
+
+# Loop to run tests
+for (( i=0; i<RUN_TIMES; i++ ))
+do
+    # Create a kind cluster
+    kind create cluster --name=e2e
+
+    # Load test images into the kind cluster
+    kind load docker-image --name=e2e quay.io/open-cluster-management/registration-operator:e2e
```
Comment (qiujian16) [inline hack/e2e_flaky_error_test.sh:26]: we do not need to hard code the image tag.

Comment (xuezhaojun) [reply hack/e2e_flaky_error_test.sh]: Agree and updated.

**Pull request merged**